upgrader.go

```diff
@@ -235,26 +221,16 @@ func (u *Upgrader) Stop() {
 		// Interrupt any running Upgrade(), and
 		// prevent new upgrade from happening.
 		close(u.stopC)
-		u.upgradeSock.Close()
-
-		// Make sure exitC is closed if no upgrade was running.
 		u.upgradeSem <- struct{}{}
+		u.upgradeSock.Close()
```

Comment:
I assume this is why your test was hanging? Good catch! I wonder if you should submit a bug report upstream?

Reply:
This ordering chance is unrelated to the test hang, it just felt more appropriate for this to happen within the upgrade semaphore. In practice, I don't think closing it in either place is incorrect.

Also, the `upgradeSock` construct is unique to tableroll, tableflip doesn't use any unix sockets.